react/react_introduction_rewrite.md

```diff
@@ -0,0 +1,61 @@
+
+Generally speaking, A JavaScript library is a collection of code that's designed to make our lives easier, not to be confused with frameworks however as you'll be reading the differences later on in the lesson. A JavaScript Library contains pre-written code "snippets" that can be reused/reimplemented in our own ways to make building things a little bit easier for ourselves.
+
+Since we know that ReactJS is used for building user interfaces. We can come to the conclusion that ReactJS is a JavaScript Library in which their pre-written code base help us build user interfaces easier and quicker.
+
+### Why Cover React?
+
+React is one of the most powerful, widely used frontend frameworks.
+
+The landscape for frontend frameworks has been changing a lot over the last few years, so it is understandable to be worried about choosing the "wrong" one.
+[This article](https://stackoverflow.blog/2018/01/11/brutal-lifecycle-javascript-frameworks/) shows the recent development of frontend frameworks well. Once you start diving deeper into a framework, you will begin to love it. It makes your code easily scalable, more readable, and possibly a thousand times more efficient (just our modest estimation).
+
+Just to name a few reasons on why to learn React.js:
+
+- Reusability of components
+- Well supported due to its popularity
+- React is not opinionated, which means that it won't force you to follow any specific design patterns, project organizational structure, or logic. It's all up to you.
+- Smaller learning curve, especially when you already have a good grasp of JavaScript and HTML from our previous lessons.
+
+If you need more convincing, the recent [2022 StackOverflow Developer Survey](https://survey.stackoverflow.co/2022/#section-most-popular-technologies-web-frameworks-and-technologies) should be reason enough to get excited about learning React. Still unsure? Check out [this article](https://laptrinhx.com/top-10-reasons-why-you-should-learn-react-right-now-3718263995/).
+
+_Convinced and excited? Great, because it's going to change your life! Let's dive right into it._
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1.  If you haven't already, browse through the <a href="https://reactjs.org/"> React Website </a>. Don't go too indepth and dive into documentation however do read the introduction/homepage to gage an idea of how React works.
+
+2.  Read this <a href="https://www.freecodecamp.org/news/the-difference-between-a-framework-and-a-library-bd133054023f/"> FreeCodeCamp article </a> discussing the differences between a Javascript Library and a Framework
+
+3.  Lastly, skim this <a href="https://www.peerbits.com/blog/reasons-to-choose-reactjs-for-your-web-development-project.html">in-depth article</a> which explains the major benefits of using React.
```

Comment:
Can we have markdown formatted links here?

And I think this article could be a great assignment: https://blog.risingstack.com/the-history-of-react-js-on-a-timeline/

Reply:
Markdown formatted links have been added!